pkg/pillar/cmd/zedrouter/appcontainer.go

```diff
@@ -101,3 +107,42 @@ func appContainerGetStats(ipAddr net.IP) (types.AppContainerMetrics, error) {
 	// XXX collect container stats for each container with the docker API endpoint ipAddr
 	return acMetrics, nil
 }
+
+func appChangeContainerStatsACL(newIPAddr, oldIPAddr net.IP) {
+	if oldIPAddr != nil {
+		// remove the previous installed blocking ACL
+		appConfigContainerStatsACL(oldIPAddr, true)
+	}
+	if newIPAddr != nil {
+		// install the App Container blocking ACL
+		appConfigContainerStatsACL(newIPAddr, false)
+	}
+}
+
+func appConfigContainerStatsACL(appIPAddr net.IP, isRemove bool) {
+	var action string
+	if isRemove {
+		action = "-D"
+	} else {
+		action = "-A"
+	}
+	// install or remove the App Container Stats blocking ACL
+	// This ACL blocks the other Apps accessing through the same subnet to 'appIPAddr:DOCKERAPIPORT'
+	// in TCP protocol, and only allow the Dom0 process to query the App's docker stats
+	// - this blocking is only possible in the 'raw' table and 'PREROUTING' chain due to the marking
+	//   is done in the 'mangle' of 'PREROUTING'
+	// - this blocking ACL does not block the Dom0 access to the above TCP endpoint on the same
+	//   subnet. This is due to the IP packets from Dom0 to the internal bridge entering the linux
+	//   forwarding through the 'OUTPUT' chain
+	// - this blocking does not seem to work if further matching to the '--physdev', so the drop action
+	//   needs to be at network layer3
+	// - XXX currently the 'drop' mark of 0xffffff on the flow of internal traffic on bridge does not work,
+	//   later it may be possible to change below '-j DROP' to '-j MARK' action
+	err := iptables.IptableCmd("-t", "raw", action, "PREROUTING", "-d", appIPAddr.String(), "-p", "tcp", "-m", "tcp",
```

Comment:
What does -m tcp add to the check?

Reply:
this is for the '--dport' match i assume.

Reply:
taking example such as: https://www.digitalocean.com/community/tutorials/how-to-set-up-a-firewall-using-iptables-on-ubuntu-14-04
-A INPUT -p tcp -m tcp --dport 22 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 80 -j ACCEPT
but I haven't tried without the '-m tcp' though

Reply:
ok, i tried it without the '-m tcp', it also works (blocks the other App). i'll remove that